### PR TITLE
Change some network requests to be asynchronous

### PR DESCRIPTION
**Pull request by barijaona**
Done for : 
- login to Open reader server (get client auth token)
- get T token

First step for issues #344 and #724

### CODE REVIEW

**Comment by josh64x2** [comment]
When I run this with invalid credentials, I get the following log output:

```
2017-03-26 16:24:35.847027 Vienna[96407:784488] GoogleReader.m:941 -[GoogleReader loadSubscriptions:] Firing directly
2017-03-26 16:24:35.847078 Vienna[96407:784488] GoogleReader.m:947 -[GoogleReader loadSubscriptions:] Token not available, registering for notification
2017-03-26 16:24:37.775064 Vienna[96407:784488] GoogleReader.m:264 -[GoogleReader authenticateForRequest:] Start first authentication...
2017-03-26 16:24:37.795669 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.795862 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.796003 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.796287 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.796402 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.796540 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.796825 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.796959 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.797085 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.797396 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.797485 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.797600 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.797940 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.798039 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.798155 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.798429 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.798527 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.798648 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.798983 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.799105 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.799243 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.799534 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.799637 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.799718 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.799914 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.799980 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.800056 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.800252 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.800319 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.800435 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.800766 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.800919 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.801077 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.801369 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.801446 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:37.801524 Vienna[96407:784488] GoogleReader.m:173 -[GoogleReader authenticateForRequest:] Another instance is authenticating...
2017-03-26 16:24:38.178567 Vienna[96407:784527] [] __nw_connection_get_connected_socket_block_invoke 10 Connection has no connected handler
```

On the positive side of things, Vienna no longer hangs with invalid credentials (free trial expired) 😄 

```
2017-03-26 16:24:39.642691 Vienna[96407:784488] [strongRequest responseStatusCode] = 403
2017-03-26 16:24:39.642843 Vienna[96407:784488] [strongRequest responseHeaders] = {
    Connection = close;
    "Content-Length" = 24;
    "Content-Type" = "text/plain; charset=UTF-8";
    Date = "Sun, 26 Mar 2017 05:24:26 GMT";
    Server = nginx;
    "X-BQ-LoginErrorReason" = FreeTrialExpired;
}
```

**Comment by barijaona** [comment]
It's just some remnants of my debugging code. It is done with `Llog`, so it will not do anything in Deployment builds.
If you want, I can remove it but I prefer keeping it until we finish moving away from ASIHTTPRequest.

**Comment by josh64x2** [comment]
I was thinking more about waiting for the first authentication to complete before calling `-(void)authenticateForRequest:(ASIHTTPRequest *)clientRequest` again.

**Comment by barijaona** [comment]
As the requests are launched on multiple threads by RefreshManager, one has to check that the first request is completed, otherwise subsequent requests must be queued.
I agree I will have to rename some methods (especially this one) to reflect more accurately their role and behavior.

File: src/GoogleReader.m [hunk 1]
```diff
@@ -51,27 +49,31 @@
...
 
 @implementation GoogleReader
 
 @synthesize localFeeds;
-@synthesize token;
+@synthesize tToken;
 @synthesize clientAuthToken;
-@synthesize tokenTimer;
-@synthesize authTimer;
+@synthesize tTokenTimer;
```
Comment (josh64x2) [inline src/GoogleReader.m]: Oh I should have said this earlier, but properties are auto-synthesized now. We can clean it up in the future

**Comment by barijaona** [comment]
Will have to revisit it, I've noticed problems getting a T token in certain circumstances.

**Pull request merged**